### PR TITLE
core: fix project counter display on index page

### PR DESCRIPTION
**Pull request by mwasilew**
Fixes: #709

Signed-off-by: Milosz Wasilewski <milosz.wasilewski@linaro.org>

### CODE REVIEW

**Comment by chaws** [comment]
I think we have another minor bug related to this:

```python
1. class ProjectManager(models.Manager):
2.     def accessible_to(self, user):
3.        if user.is_superuser or user.is_staff:
4.             return self.all()
5.         else:
6.             groups = Group.objects.filter(members__id=user.id).only('id')
7.             return self.filter(Q(group__in=groups) | Q(is_public=True))
```

Take line 6 `Group.objects.filter(members__id=user.id)`:
* for user = AnonymousUser you get a query like 

```sql
SELECT "core_group"."id" 
FROM "core_group" LEFT OUTER JOIN "core_groupmember" ON (
        "core_group"."id" = "core_groupmember"."group_id")
WHERE "core_groupmember"."user_id" IS NULL ORDER BY "core_group"."slug" ASC
```

* whereas for user = User (not admin nor staff) the query would have `IS NULL` replaced by something like `ID IN (...)`

Because of `LEFT OUTER JOIN` behavior, it'll being groups that don't have any members related to it, which I think is wrong!

**Comment by mwasilew** [comment]
@chaws I think it's OK as long as the project is public. Am I missing something?

**Comment by chaws** [comment]
@mwasilew I did some more digging:

Let:
```python
# group without members
gwom = Group.objects.filter(members__id__isnull=True).first()
public_project_gwom = gwom.projects.filter(is_public=True).first()
private_project_gwom = gwom.projects.filter(is_public=False).first()

# group with memebers
gwm = Group.objects.filter(members__id__isnull=False).first()
public_project_gwm = gwm.projects.filter(is_public=True).first()
private_project_gwm = gwm.projects.filter(is_public=False).first()
```

I tested the following conditions:

| group-project/user   | Anonymous User                             | Member User                                | Not-a-member User                              |
|----------------------|--------------------------------------------|--------------------------------------------|------------------------------------------------|
| gwom                 | sees group: yes<br>sees project count: all | N/A                                        | sees group: yes*<br>sees project count: public |
| gwm                  | sees group: no                             | sees group: yes<br>sees project count: all | sees group: yes*<br>sees project count: public |
| public_project_gwom  | sees project: yes                          | N/A                                        | sees project: yes                              |
| private_project_gwom | sees project: no                           | N/A                                        | sees project: no                               |
| public_project_gwm   | sees project: yes                          | sees project: yes                          | sees project: yes                              |
| private_project_gwm  | sees project: no                           | sees project: yes                          | sees project: no                               |

The `yes*` means that group is only visible because there's at least one public project within.

It's just weird that an Anonymous user can still see a group's home page with a list of all projects (private ones included!), whereas a user that's logged in cannot. We can discuss that in the future though.

File: squad/core/models.py [hunk 1]
```diff
@@ -46,9 +46,11 @@ def accessible_to(self, user):
         projects = Project.objects.accessible_to(user)
         project_ids = [p.id for p in projects]
         group_ids = set([p.group_id for p in projects])
+        if not isinstance(user, AnonymousUser):
+            group_ids = group_ids | set([g.id for g in Group.objects.filter(members__in=[user])])
         return self.filter(
-            Q(id__in=group_ids) | Q(members__id=user.id)
-        ).order_by('slug').annotate(
+            id__in=group_ids
+        ).distinct().order_by('slug').annotate(
```
Comment (chaws) [inline squad/core/models.py:53]: I tried it locally, `distinct()` is unnecessary as you already removed the join `members__id=user.id`.

**Comment by chaws** [comment]
It looks good, tricky bug to catch. I'm merging this now, we can work on a new PR to fix access issues I mentioned above

**Pull request merged**